tests/testsuite.src/run_misc.at

```diff
@@ -14570,3 +14570,32 @@ TST-DECIMAL IS < ZERO-DECIMAL
 
 
 AT_CLEANUP
+
+AT_SETUP([Decimal constants working after sub-program call])
```

Comment:
If I'm not wrong this is only happening on `INITIAL` sub-programs, as only those redo the init.
In this case please adjust to `decimal constants and INITIAL nested programs`, and add another empty line before AT_SETUP above

Reply:
Note that this bug does not only affect "nested" programs, but also consecutive programs in the same source file.

Reply:
Good to know, then `decimal constants with multiple INITIAL programs in same source unit` would be better, but that's both too long and doesn't read that well...

Actually your test example doesn't use nested, but it seems to be good to also test that.
What do you think about copying the source to have that tested, either as separate test or in the same test (then possibly moving prog2 out to a copybook /  use conditional compilation)?